src/repr/src/row/encoding.rs

```diff
@@ -998,4 +1002,34 @@ mod tests {
             Ok(())
         );
     }
+
+    fn scalar_type_columnar_roundtrip(scalar_type: ScalarType) {
+        use mz_persist_types::parquet::validate_roundtrip;
+        let mut rows = Vec::new();
+        for datum in scalar_type.interesting_datums() {
+            rows.push(Row::pack(std::iter::once(datum)));
+        }
+
+        // Non-nullable version of the column.
+        let schema = RelationDesc::empty().with_column("col", scalar_type.clone().nullable(false));
+        for row in rows.iter() {
+            assert_eq!(validate_roundtrip(&schema, row), Ok(()));
+        }
+
+        // Nullable version of the column.
+        let schema = RelationDesc::empty().with_column("col", scalar_type.nullable(true));
+        rows.push(Row::pack(std::iter::once(Datum::Null)));
+        for row in rows.iter() {
+            assert_eq!(validate_roundtrip(&schema, row), Ok(()));
+        }
+    }
+
+    proptest! {
+       #[test]
+       #[cfg_attr(miri, ignore)] // too slow
+        fn all_scalar_types_columnar_roundtrip(scalar_type in any::<ScalarType>() ) {
+            // The proptest! macro interferes with rustfmt.
```

Comment:
FWIW I find that [closure style](https://altsysrq.github.io/rustdoc/proptest/latest/proptest/macro.proptest.html#closure-style-invocation) helps keep the unformattable code small!